### PR TITLE
(PUP-7042) Mark strings in file_serving

### PR DESCRIPTION
**Pull request by Magisus**
This commit marks user-facing error and info strings in
`lib/puppet/file_server/*` for translation.

### CODE REVIEW

**Comment by Magisus** [comment]
Closing and reopening to fix Travis.

**Comment by Magisus** [comment]
Closing and reopening to fix CI.

File: lib/puppet/file_serving/fileset.rb [hunk 1]
```diff
@@ -47,8 +47,8 @@ def initialize(path, options = {})
       initialize_from_hash(options)
     end
 
-    raise ArgumentError.new("Fileset paths must exist") unless valid?(path)
-    raise ArgumentError.new("Fileset recurse parameter must not be a number anymore, please use recurselimit") if @recurse.is_a?(Integer)
+    raise ArgumentError.new(_("Fileset paths must exist")) unless valid?(path)
+    raise ArgumentError.new(_("Fileset recurse parameter must not be a number anymore, please use recurselimit")) if @recurse.is_a?(Integer)
```
Comment (MosesMendoza) [inline lib/puppet/file_serving/fileset.rb]: Can you add a note that `recurse` is an app parameter that isn't translated (same for recurselimit)?

File: lib/puppet/file_serving/content.rb [hunk 2]
```diff
@@ -30,7 +30,7 @@ def collect(source_permissions = nil)
   def content
     unless @content
       # This stat can raise an exception, too.
-      raise(ArgumentError, "Cannot read the contents of links unless following links") if stat.ftype == "symlink"
+      raise(ArgumentError, _("Cannot read the contents of links unless following links")) if stat.ftype == _("symlink")
```
Comment (MosesMendoza) [inline lib/puppet/file_serving/content.rb]: I don't think "symlink" should be wrapped here - that's going to be a string returned by ruby

File: lib/puppet/file_system/file_impl.rb [hunk 3]
```diff
@@ -14,7 +14,7 @@ def assert_path(path)
     if path.respond_to?(:to_str)
       Pathname.new(path)
     else
-      raise ArgumentError, "FileSystem implementation expected Pathname, got: '#{path.class}'"
+      raise ArgumentError, _("FileSystem implementation expected Pathname, got: '#{path.class}'")
```
Comment (MosesMendoza) [inline lib/puppet/file_system/file_impl.rb]: Looks like from here on in the PR interpolations are incorrect.

Comment (Magisus) [inline lib/puppet/file_system/file_impl.rb]: Ah, another subdirectory accidentally lumped in the wrong PR. Not going to split it out, I'll just fix it up here.

**Comment by Magisus** [comment]
Updated.

**Pull request merged**